src/test/custom_nodes/node_choose_maximum.cpp

```diff
@@ -0,0 +1,163 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <algorithm>
+#include <cstddef>
+#include <cstring>
+#include <iostream>
+#include <limits>
+#include <string>
+#include <vector>
+
+#include "../../custom_node_interface.h"
+
+static const size_t dummyInputSize = 10;
+
+extern "C" {
+enum METHOD {
```

Comment:
Since this is C++, we could use enum class

Reply:
Not because I am passing this in params which are not of the type METHOD. Otherwise I would have to add conversion function - not worthy IMO.